docx_editor/track_changes.py

```
@@ -335,6 +428,71 @@ def __repr__(self) -> str:
+    if not isinstance(target, SearchResult):
+        return target, paragraph, occurrence
+    redundant = [
+        name
+        for name, given in (("paragraph=", paragraph is not None), ("occurrence=", occurrence is not None))
+        if given
+    ]
+    if redundant:
+        raise ValueError(
+            f"{ctx}{arg} is a SearchResult, which already pins the paragraph ({target.paragraph_ref!r}) "
+            f"and the occurrence ({target.paragraph_occurrence}) — drop {' and '.join(redundant)}"
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)